repo/maintenance/maintenance_run.go

```diff
@@ -456,9 +457,20 @@ func runFullMaintenance(ctx context.Context, runParams RunParameters, safety Saf
 		return errors.Wrap(err, "error cleaning up epoch manager")
 	}
 
+	if err := runTaskCleanup(ctx, runParams.rep, s); err != nil {
+		return errors.Wrap(err, "error running cleanup task")
+	}
```

Comment:
Do we want to make cleanup optional? For instance if the user wanted to handle blob lifecycles outside of kopia? The S3 implementation doesn't have the equivalent cleanup, IIRC - delete markers and old versions stick around until cleaned up externally. Perhaps if we clean up for AZ we should be cleaning up for S3 as well?

Reply:
Could make it optional 👍

Reply:
Thinking about it some more I'm not so sure about making it optional. As mentioned in the other comment the files will only truly be cleaned up if there is a BLM setup, which is enabled by default in Azure but could be disabled if the user chose to. Having a flag here would make it different to non-ransomware Azure & both variations of S3 where the blobs are delete markered and eventually cleaned up if there is a BLM present.